src/pages/PoolDetails/PoolDetailsHeader.test.tsx

```diff
@@ -0,0 +1,32 @@
+import userEvent from '@testing-library/user-event'
+import { act, render, screen } from 'test-utils/render'
+
+import { PoolDetailsHeader } from './PoolDetailsHeader'
+
+describe('PoolDetailsHeader', () => {
+  const mockProps = {
+    chainId: 1,
+    poolAddress: '0x88e6a0c2ddd26feeb64f039a2c41296fcb3f5640',
+    token0: { id: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48', symbol: 'USDC' },
+    token1: { id: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2', symbol: 'WETH' },
+    feeTier: 500,
+    toggleReversed: jest.fn(),
+  }
+
+  it('renders header text correctly', () => {
+    render(<PoolDetailsHeader {...mockProps} />)
+
+    expect(screen.getByText(/Explore/i)).toBeInTheDocument()
+    expect(screen.getByText(/Pool/i)).toBeInTheDocument()
+    expect(screen.getByText(/USDC \/ WETH \(0x88e6...5640\)/i)).toBeInTheDocument()
+    expect(screen.getByText('0.05%')).toBeInTheDocument()
```

Comment:
you might consider a snapshot test here instead

```
  expect(asFragment()).toMatchSnapshot()
```

Reply:
I generally querying for specific elements as opposed to a more broad snapshot match as it's much easier to glance over a breaking change when updating snapshots vs having to make sure the output looks as you intended it.

Reply:
I get that, but I think text presence is much less likely to break here relative to styling. Snapshots get you the style safety *and* the text safety.

Reply:
I've also added snapshot tests to cover both scenarios